Feature/hosts description

jlariza opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Fixes:  #3348 
Hosts object have two main fields, name and description. The name tends to be an IP address that is not very explicit. However, all the hosts list do not display their description, only their name so they are very confusing.

This Branch adds the description fields to the hosts list so they are easily identifiable with no extra effort.
<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - UI

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 4.0.0
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

jlariza committed:
added description to host list - #3348

Signed-off-by: Jose Ariza <l.jlac001@gmail.com>

Signed-off-by: Jose Ariza <l.jlac001@gmail.com>

Signed-off-by: Jose Ariza <l.jlac001@gmail.com>

Signed-off-by: Jose Ariza <l.jlac001@gmail.com>

Signed-off-by: Jose Ariza <l.jlac001@gmail.com>

Signed-off-by: Jose Ariza <l.jlac001@gmail.com>

cdvv7788 commented:
@ryanpetrello How can we check the failures in the build? Are there any instructions for that?

ryanpetrello commented:
@cdvv7788 the current build failures are an issue w/ our zuul CI that we're looking into (as we speak).

ryanpetrello commented:
recheck

wenottingham commented:
What sources actually populate `description`? Is this only ever manually set?

shanemcd commented:
recheck

shanemcd commented:
recheck

cdvv7788 commented:
@wenottingham I am not sure if this is populated only manually. The changes are only for visualization tho, so there shouldn't be an issue even if they are inserted somewhere else.

wenottingham commented:
I understand; I just wonder about always including a column if none of AWX's inventory sources populate it.

cdvv7788 commented:
@wenottingham So we can include it conditionally. Would that be better? It should not be that hard to check if the hosts have description if they are already fetched.

lijok commented:
@wenottingham @cdvv7788 to chime in, i know import inventory from a script sets description to "imported" https://github.com/ansible/awx/blob/devel/awx/main/management/commands/inventory_import.py#L667

mabashian commented on awx/ui/client/src/inventories-hosts/hosts/hosts.partial.html:
```diff
@@ -64,17 +65,22 @@
                                     </div>
                                 </div>
-                                <div class="List-tableCell inventory_name-column col-lg-4 elllipsis">
+                                <div class="List-tableCell name-column InventoryManage-breakWord col-lg-3">
```
@jlariza I don't think we want the description to be a link - we don't have that pattern anywhere else in the application.  I think you'll want this code to look like:

```
                                <div class="List-tableCell name-column InventoryManage-breakWord col-lg-3">
                                    {{host.description }}
                                </div>
```

mabashian commented on awx/ui/client/src/inventories-hosts/inventories/related/hosts/related-host.list.js:
```diff
@@ -64,7 +64,17 @@ export default ['i18n', function(i18n) {
+            description: {
+                key: true,
+                label: i18n._('Description'),
+                uiSref: ".edit({inventory_id: host.inventory_id,host_id: host.id})",
```
See above - this shouldn't be a link.  You can get rid of `key`, `uiSref`, `ngClass`, `dataHostId`, `dataType`, and `class`.

mabashian commented on awx/ui/client/src/inventories-hosts/inventories/related/hosts/related-host.list.js:
```diff
@@ -64,7 +64,17 @@ export default ['i18n', function(i18n) {
                 uiSref: ".edit({inventory_id: host.inventory_id,host_id: host.id})",
                 ngClass: "{ 'host-disabled-label': !host.enabled }",
-                columnClass: 'col-lg-5 col-md-4 col-sm-8 col-xs-7',
+                columnClass: 'col-lg-3 col-md-3 col-sm-8 col-xs-7',
```
Change to `columnClass: 'col-lg-3 col-md-6 col-sm-8 col-xs-7',`

mabashian commented on awx/ui/client/src/inventories-hosts/inventories/related/hosts/related-host.list.js:
```diff
@@ -64,7 +64,17 @@ export default ['i18n', function(i18n) {
+                label: i18n._('Description'),
+                uiSref: ".edit({inventory_id: host.inventory_id,host_id: host.id})",
+                ngClass: "{ 'host-disabled-label': !host.enabled }",
+                columnClass: 'col-lg-3 col-md-3 col-sm-8 col-xs-7',
```
Change to `columnClass: 'd-none d-lg-flex col-lg-3'`

mabashian commented on awx/ui/client/src/inventories-hosts/inventories/related/hosts/related-host.list.js:
```diff
@@ -74,13 +84,13 @@ export default ['i18n', function(i18n) {
                 nosort: true,
                 showDelete: true,
-                columnClass: 'd-none d-md-flex List-tableCell col-lg-5 col-md-4'
+                columnClass: 'd-none d-md-flex List-tableCell col-lg-3 col-md-3'
```
Change to `columnClass: 'd-none d-lg-flex List-tableCell col-lg-3'`

mabashian commented on awx/ui/client/src/inventories-hosts/inventories/related/hosts/related-host.list.js:
```diff
@@ -74,13 +84,13 @@ export default ['i18n', function(i18n) {
         fieldActions: {
 
-            columnClass: 'col-lg-2 col-sm-4 col-xs-5 text-right',
+            columnClass: 'col-lg-2 col-sm-3 col-xs-5 text-right',
```
Change to `columnClass: 'col-lg-3 col-md-6 col-sm-4 col-xs-5 text-right',`

mabashian commented on awx/ui/client/src/inventories-hosts/hosts/hosts.partial.html:
```diff
@@ -43,9 +43,10 @@
-                            <div class="List-tableHeader" base-path="hosts" collection="hosts" dataset="host_dataset" column-sort="" column-field="name" column-iterator="host" column-no-sort="undefined" column-label="Name" column-custom-class="col-lg-4" query-set="host_queryset"></div>
-                            <div class="List-tableHeader" base-path="hosts" collection="hosts" dataset="host_dataset" column-sort="" column-field="inventory" column-iterator="host" column-no-sort="undefined" column-label="Inventory" column-custom-class="d-none d-sm-flex col-sm-4 ellipsis" query-set="host_queryset"></div>
-                            <div class="List-tableHeader List-tableHeader--actions actions-column col-lg-4 text-right" translate>Actions</div>
+                            <div class="List-tableHeader" base-path="hosts" collection="hosts" dataset="host_dataset" column-sort="" column-field="name" column-iterator="host" column-no-sort="undefined" column-label="Name" column-custom-class="col-lg-3" query-set="host_queryset"></div>
```
We'll want to apply the same Bootstrap grid layout classes (`col-...`) to these elements.  See the proposed changes to `related-host.list.js` for context.  For `lg` sized screens we should show all 4 columns but we want to hide groups/description on smaller screens.

jlariza commented:
Hi @mabashian, thanks for the recommendations. I just applied them. I had to make some modifications in order to make it look good on mobile.

Please, check it again and let me know if something is missing. Thank you

mabashian commented:
@jlariza noticing a couple of things when taking a look at the most recent changes:

1) I doesn't seem like ellipse-ifying the description is working as expected on `#/hosts`.  The column alignment gets thrown off when the description is wider than the column.
<img width="1496" alt="Screen Shot 2019-07-02 at 9 21 04 AM" src="https://user-images.githubusercontent.com/9889020/60516480-d5b10680-9cab-11e9-96f1-c454266237c2.png">

2) If we're going to ellipse the description on `#/hosts` we should do the same thing for the hosts list within an inventory `#/inventories/inventory/:id/hosts`
<img width="1511" alt="Screen Shot 2019-07-02 at 9 27 01 AM" src="https://user-images.githubusercontent.com/9889020/60516621-2294dd00-9cac-11e9-97ff-0d4007a5f3d0.png">

We'll also need you to squash down your commits before we get this merged in.

jlariza commented:
@mabashian Done. I removed the ellipsis class; it made no sense to cut the description If what we want is to ease each host identification.

I already squashed the commits. Thank you

mabashian commented:
recheck

mabashian commented:
@jlariza I branched off of your branch and made a few extra changes.  That PR can be found here: https://github.com/ansible/awx/pull/4304.  If it's alright I'd like to move forward with these changes in that PR.  Thank you for your work on this feature and I apologize that it's taken this long to get the ball rolling.

jlariza commented:
@mabashian sure, go Ahead. I checked the changes you made and they make sense. Thank you!

mabashian commented:
@jlariza https://github.com/ansible/awx/pull/4304 has been merged - thanks for your efforts on that.  We appreciate it!
